making rule trigger deletion great again

lauracorssac opened the pull request.

lucio-lpf commented:
It's a good practice to left just 1 commit for PR. That way it will be easier to see things in master branch. If you want to incorporate more code in an existed commit, look for - git commit --amend - command. If you want to att a branch commit, look for - git push --force-with-lease - command.

lauracorssac commented:
> It's a good practice to left just 1 commit for PR. That way it will be easier to see things in master branch. If you want to incorporate more code in an existed commit, look for - git commit --amend - command. If you want to att a branch commit, look for - git push --force-with-lease - command.

I agree with you that it is easier to find and remove buggy branches merged into master if they are only composed by a single commit. However, during the development in a single branch, I think it is better to have many commits. I think it helps you to organize yourself, find when you did a wrong commit and also, if you don't understand a piece of code and execute git blame, you can see the commits that modified it. If they are expressive and not so generic, you may be able solve your problem quicker. 

Maybe, we can, instead of merging a PR, merge it squashed, although I am not sure if by doing that, we can still see the history of commits done before executing this command when git blaming a piece of code in master.

lucio-lpf commented:
> > It's a good practice to left just 1 commit for PR. That way it will be easier to see things in master branch. If you want to incorporate more code in an existed commit, look for - git commit --amend - command. If you want to att a branch commit, look for - git push --force-with-lease - command.
> 
> I agree with you that it is easier to find and remove buggy branches merged into master if they are only composed by a single commit. However, during the development in a single branch, I think it is better to have many commits. I think it helps you to organize yourself, find when you did a wrong commit and also, if you don't understand a piece of code and execute git blame, you can see the commits that modified it. If they are expressive and not so generic, you may be able solve your problem quicker.
> 
> Maybe, we can, instead of merging a PR, merge it squashed, although I am not sure if by doing that, we can still see the history of commits done before executing this command when git blaming a piece of code in master.

Tasks should not be "so generic". If you are working too much on a task it should be broken into other tasks (more PRs). 

See this print. The commits in green color tell me some really useful information. The red one does not.

<img width="1108" alt="Captura de Tela 2020-04-02 às 10 31 25" src="https://user-images.githubusercontent.com/8583169/78227690-ee19b000-74cd-11ea-982f-c09167716ebb.png">

 I agree that sometimes it is ok to have more than 1 commit for PR (if the task is too big and no break suitable), but in the case of this picture and on this PR I don't think it is.

lauracorssac commented:
> > > It's a good practice to left just 1 commit for PR. That way it will be easier to see things in master branch. If you want to incorporate more code in an existed commit, look for - git commit --amend - command. If you want to att a branch commit, look for - git push --force-with-lease - command.
> > 
> > 
> > I agree with you that it is easier to find and remove buggy branches merged into master if they are only composed by a single commit. However, during the development in a single branch, I think it is better to have many commits. I think it helps you to organize yourself, find when you did a wrong commit and also, if you don't understand a piece of code and execute git blame, you can see the commits that modified it. If they are expressive and not so generic, you may be able solve your problem quicker.
> > Maybe, we can, instead of merging a PR, merge it squashed, although I am not sure if by doing that, we can still see the history of commits done before executing this command when git blaming a piece of code in master.
> 
> Tasks should not be "so generic". If you are working too much on a task it should be broken into other tasks (more PRs).
> 
> See this print. The commits in green color tell me some really useful information. The red one does not.
> 
> <img alt="Captura de Tela 2020-04-02 às 10 31 25" width="1108" src="https://user-images.githubusercontent.com/8583169/78227690-ee19b000-74cd-11ea-982f-c09167716ebb.png">
> 
> I agree that sometimes it is ok to have more than 1 commit for PR (if the task is too big and no break suitable), but in the case of this picture and on this PR I don't think it is.

Hmm. But let's suppose you do all of your work and execute it. You are sure it works and then commits. Then you think of improving the code. You do some optimizations and refactoring, let's suppose. In my view, it is good to separate this improvements in a new commit, because even if you execute it, you can be missing something or something could go wrong in this process. Later you can compare the execution of this two commits and see if the mistake was introduced by the feature itself or by its enhancing.. Ok, if you have lots of tests implemented, this separation maybe doesn't help so much, but it isn't the reality of many big projects.

In my case "removing logs and fixes in texts" is that I removed logs and changed the texts. Maybe by the removal of one of the logs I could change something in the execution. Let's suppose I forgot to execute the program in a certain way and later on I realize it. It could be due to the change of the texts and the removal of the logs. Hahaha. It happened to me once. It was C, but it happened hahaha. 

So, to end this discussion, I think there are pros and cons in both ways. I am for many commits, but if the whole team decides to implement in another way, of course I wouldn't matter to change.. If it is already a guideline, I didn't know that. 😧

schneijan commented on src/main/java/org/citopt/connde/web/rest/RestComponentFilterController.java:
```diff
@@ -37,9 +38,37 @@
+    @ApiResponses({@ApiResponse(code = 200, message = "Success")})
+    public ResponseEntity<List<Rule>> getRulesByRuleTriggerID(@PathVariable(value = "id") @ApiParam(value = "ID of the rule trigger", example = "5c97dc2583aeb6078c5ab672", required = true) String ruleTriggerId) {
+
+        List<Rule> rules = ruleRepository.findAll();
```
Would it be possible to make use of `RuleTrigger ruleTrigger = ruleTriggerRepository.findOne(ruleTriggerId);` here? This would ensure that the deleting user really has access to the rule trigger for which the ID has been provided.

lauracorssac replied:
Hmm I understand. I think I assumed the user would always have access to it as he/she is the one who is deleting. But I agree that it is good to check so that the function can be more secure and can be used in other contexts!

schneijan replied:
Yes, you are right, it's not a (big) security risk. We just should avoid that a user is able to check what rules/triggers a foreign user may have defined or which rules are shared with this user. :-)
